src/controller/CommissioningDelegate.h

```diff
@@ -71,6 +75,13 @@ enum CommissioningStage : uint8_t
     kNeedsNetworkCreds,
 };
 
+enum ICDRegistrationStrategy
+{
+    kIgnore,         ///< Do not check whether the device is an ICD during commissioning
+    kBeforeComplete, ///< Do commissioner self-registration or external controller registration,
```

Comment:
should we create two strategy for self-registration and external controller registration? kBeforeComplete is clear enough to tell us what happens for this strategy.

Reply:
In my mind, from the view of the CHIPDeviceController class and the commissionee, there is no difference between self registration and external controller registration, since all ICD registration commands are send from *this* controller and the only difference is who will control the device after commission.

In later PRs, we may introduce a few delegates for actually handle the keys for ICD subscription.

Do you have any suggestions on the name of this value?